src/nlsolve/utils.jl

```diff
@@ -120,6 +120,92 @@ function build_nlsolver(alg,nlalg::Union{NLFunctional,NLAnderson,NLNewton},u,upr
     nlcache)
 end
 
+mutable struct SuperDAEJacobianWrapper{F,pType,duType,uType,gammaType,tmpType,uprevType,tType} <: Function
+  f::F
+  p::pType
+  tmp_du::duType
+  tmp_u::uType
+  γ::gammaType
+  tmp::tmpType
+  uprev::uprevType
+  t::tType
+  function SuperDAEJacobianWrapper(f,p,γ,tmp,uprev,t)
+    X = eltype(uprev)
+    N = ForwardDiff.chunksize(ForwardDiff.Chunk(uprev))
+    tmp_du = similar(uprev, ForwardDiff.Dual{Nothing,X,N})
+    tmp_u  = similar(uprev, ForwardDiff.Dual{Nothing,X,N})
+    new{typeof(f),typeof(p),typeof(tmp_du),typeof(tmp_u),typeof(γ),typeof(tmp),typeof(uprev),typeof(t)}(f,p,tmp_du,tmp_u,γ,tmp,uprev,t)
+  end
+end
+
+function (m::SuperDAEJacobianWrapper)(out,x)
```

Comment:
This would be a million times more clear if it just had a good name. `DAEResidualJacobianWrapper` is what it is: it embeds all previous information to give the Jacobian for the residual calculation. I'd add a comment as to what this is actually calculating, i.e. why this corresponds to `f(x) = 0`